### DISCUSSION TITLE
Unix dependency installation

### DISCUSSION THREAD

**Discussion by poetaman** [poetaman/arttime#23] [General]
As of writing this, GitHub doesn't allow PRs for wiki pages of repositories. You are welcome to leave your Unix flavor-specific installation commands for arttime's dependencies here. Create a bullet list like for Debian below (Linux has its own discussion page: [Linux dependency installation](https://github.com/poetaman/arttime/discussions/22)). It might also get echoed on [arttime wiki](https://github.com/poetaman/arttime/wiki).

```
### Debian
* ``zsh``: ``apt install zsh``
* ``vorbis-tools``: ``apt install vorbis-tools`` (not strictly needed as pulseaudio is preinstalled, though recommend as a fallback to have anyway)
* ``notify-send``: ``apt install libnotify-bin``
```

**Comment by grahamperrin** [comment]
```text
### FreeBSD
* ``zsh``: ``pkg install shells/zsh``
* ``notify-send``: ``pkg install devel/libnotify``
* ``paplay``: ``pkg install audio/pulseaudio``
```

Side note, for users of PulseAudio: https://github.com/poetaman/arttime/discussions/26

**Comment by poetaman** [comment]
@grahamperrin @ehaupt From upcoming release, arttime package managers for both Linux and Unixes are advised to change arttime  dependencies to a fixed set:

1. zsh or equivalent to get ``zsh`` (>=5.8)
2. vorbis-tools or equivalent to get ``ogg123``
3. libnotify/libnotify-bin or equivalent to get ``notify-send``

Arttime will still find and prefer native interface of pulseaudio or pipewire sound daemons, and other sound daemons might be detected in future. But none of the sound daemons should be arttime's dependency as they might conflict with each other or user might want to change their sound server daemon after installing arttime (and multiple sound server daemons cannot coexist happily afaik). If a known daemon is not running, arttime will always fallback to vorbis-tool's ogg123. That should be the only sound/audio dependency in package managers for arttime.

**Comment by poetaman** [comment]
@ehaupt  v1.9.8 changes dependencies:

1. ``zsh`` (>=5.7) to get arttime running
2. ``vorbis-tools`` to get sounds (needed only if pulseaudio or pipewire are not installed, though recommend as a fallback to have anyway)
3. ``notify-send`` to get desktop notifications
4. ``less`` to navigate help pages during runtime.  Arttime would fallback on ``more`` which should be preinstalled on every unix system, though ``less`` is better for user experience and recommended.

From prior version, the only two changes are: 1) zsh version requirement goes down from 5.8 to 5.7 (makes arttime more available), and 2) ``less`` is added to dependencies.

Here's the link to [v1.9.8 release notes](https://github.com/poetaman/arttime/releases/tag/v1.9.8).

**Comment by poetaman** [comment]
@ehaupt I did a fresh install of FreeBSD 14.0. It looks like we would need https://www.freshports.org/misc/terminfo-db/ as a dependency. Or else ``TERM=xterm-256color arttime`` still complains about missing capabilities (which are part of vt100 spec). The default terminal database shipped with FreeBSD 14.0 seems stale.

**Reply by ehaupt** [reply]
Will simply pulling in `misc/terminfo-db` as a runtime dependency make arttime function properly or would it require additional changes in the code?

**Reply by poetaman** [reply]
@ehaupt Simply pulling ``misc/terminfo-db`` works. If invoking via ssh, user might have to invoke arttime as ``TERM=xterm-256color arttime``, or perhaps invoke ssh like this: ``TERM=xterm-256color ssh ...``.

**Reply by ehaupt** [reply]
While we're at it, it seems like a good idea to pull in `textproc/fzf` as well.

**Reply by poetaman** [reply]
@ehaupt If it is not too large a dependency. When I try installing ``fzf`` it on my new FreeBSD 14.0 install, it seems wanting to install ``bash`` too. From the message below, it's not clear to me whether the disk space increase will be 3MB or 13MB. The route that AUR package maintainer has taken is to keep dependencies optional for his concern of disk space. IMO each platform's users have a personality/culture (as it seems from AUR maintainers views), not sure whether BSD crowd cares about additional 13MB. I believe you will make a better tradeoff/choice here than I can.

```
root@freebsd14:~ # pkg install fzf
Updating FreeBSD repository catalogue...
FreeBSD repository is up to date.
All repositories are up to date.
The following 2 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
        bash: 5.2.15
        fzf: 0.38.0

Number of packages to be installed: 2

The process will require 13 MiB more space.
3 MiB to be downloaded.

Proceed with this action? [y/N]:
```
